### PR TITLE
Add CRD validation schemata

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: hack/update-crd-validation/main.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	codecs = serializer.NewCodecFactory(scheme)
+)
+
+func init() {
+	v1beta1.AddToScheme(scheme)
+}
+
+func main() {
+	// load existing manifests from manifests/ dir
+	existingManifests, err := crdsFromDirectory("manifests")
```
Comment (deads2k) [inline hack/update-crd-validation/main.go]: I'm ok starting here.   Seems like we should just generate everything though

File: manifests/0000_05_config-operator_01_authentication.crd.yaml [hunk 2]
```diff
@@ -13,6 +14,12 @@ spec:
   subresources:
     status: {}
   versions:
-    - name: v1
-      served: true
-      storage: true
+  - name: v1
+    served: true
+    storage: true
+status:
```
Comment (deads2k) [inline manifests/0000_05_config-operator_01_authentication.crd.yaml]: I'm not clear on why we're doing this, but ok.

**Comment by deads2k** [comment]
@sttts had to tweak the generator?  I can live with that, but maybe a separate commit to make more easily pickable?

/lgtm

**Comment by damemi** [comment]
Do we still want to pursue https://github.com/kubernetes-sigs/controller-tools/pull/144? Could do everything from the makefile without a hack file and a dependency on the controller tools. I'm open to either

File: hack/update-crd-validation/main.go (hunk 1 above)
Comment (sttts) [reply hack/update-crd-validation/main.go]: what else?

Comment (sttts) [reply hack/update-crd-validation/main.go]: We have to copy this mechanism to all repos that have CRDs.

**Comment by sttts** [comment]
> @sttts had to tweak the generator? I can live with that, but maybe a separate commit to make more easily pickable?

No tweaking. I copy files where the generator expects them, and I only copy the validation part. So whatever the generator does outside, we don't care. @damemi's PR against the generator will fix this, so we can make use of more if we want to.

File: Makefile [hunk 3]
```diff
@@ -31,3 +31,14 @@ $(call add-bindata,v3.11.0,./bindata/v3.11.0/...,bindata,v311_00_assets,pkg/oper
 clean:
 	$(RM) ./cluster-config-operator
 .PHONY: clean
+
+update-codegen:
```
Comment (soltysh) [inline Makefile]: Don't you want to move that to library-go?

Comment (soltysh) [reply Makefile]: @tnozicka fyi

File: Makefile [hunk 4]
```diff
@@ -31,3 +31,14 @@ $(call add-bindata,v3.11.0,./bindata/v3.11.0/...,bindata,v311_00_assets,pkg/oper
...
+	go run ./vendor/github.com/openshift/library-go/cmd/crd-schema-gen/main.go --apis-dir vendor/github.com/openshift/api
+.PHONY: update-codegen
+
+verify-codegen: update-codegen
```
Comment (tnozicka) [inline Makefile]: this should not depend on update

File: Makefile (hunk 3 above)
Comment (tnozicka) [inline Makefile]: please rename the target to `update-codegen-crd` and you can hook it into `update-codegen: update-codegen-crd` if you want so it doesn't conflict with standart library-go targets

File: manifests/0000_05_config-operator_01_authentication.crd.yaml (hunk 2 above)
Comment (sttts) [reply manifests/0000_05_config-operator_01_authentication.crd.yaml]: fixed. That were default values that yaml marshalling added. Removed them by not going via CRD golang types.

**Comment by sttts** [comment]
image operator.

/retest

**Comment by sttts** [comment]
aws flake.

/retest

**Comment by deads2k** [comment]
nothing has merged, looks like tide doesn't see approved. merging

**Pull request merged**